File: inst/Matlab/fdaM/lambda2gcv.m
function gcv = lambda2gcv(log10lambda, argvals, y, fdParobj, ...
                          wtvec, dffactor)
%  LAMBDA2GCV smooths data using smooth_basis and returns the 
%  GCV criterion that results.  DFFACTOR is an optional multiplier of df
%  in the definition of GCV

if nargin < 4, error('Insufficient number of arguments.');  end
if nargin < 6, dffactor = 1;               end
if nargin < 5, wtvec = ones(length(y),1);  end

fdParobj = putlambda(fdParobj, 10^log10lambda);

[fdobj, df, gcv] = ...
    smooth_basis(argvals, y, fdParobj, wtvec, dffactor);

File: R/lambda2gcv.R
lambda2gcv <- function(log10lambda, argvals, y, fdParobj,
                       wtvec=rep(1,length(y)))
{
#  LAMBDA2GCV smooths data using smooth_basis and returns the
#  GCV criterion that results.  DFFACTOR is an optional multiplier of df
#  in the definition of GCV

fdParobj$lambda = 10^log10lambda

#smoothlist <- smooth.basis(argvals, y, fdParobj, wtvec, dffactor)
smoothlist <- smooth.basis(argvals, y, fdParobj, wtvec)

gcv <- smoothlist$gcv

gcv

}
